packages/lib/src/vis/tiles/TiledHeatmap.tsx

```diff
@@ -0,0 +1,82 @@
+import { clamp, range } from 'lodash';
+import { useMemo } from 'react';
+import { Color } from 'three';
+
+import { getInterpolator } from '../heatmap/utils';
+import { useFrameRendering } from '../hooks';
+import { useAxisSystemContext } from '../shared/AxisSystemContext';
+import TiledHeatmapLayer from './TiledHeatmapLayer';
+import type { TilesApi } from './api';
+import { useDataPerPixel } from './hooks';
+import type { ColorMapAttributes } from './models';
+
+interface Props extends ColorMapAttributes {
+  api: TilesApi;
+  progressive?: boolean;
+  qualityFactor?: number;
+}
+
+function TiledHeatmap(props: Props) {
+  const {
+    api,
+    progressive = true,
+    qualityFactor = 1, // 0: Lower quality, less fetch; 1: Best quality
+    ...colorMapAttrs
+  } = props;
+
+  const { visSize } = useAxisSystemContext();
+  const { xDataPerPixel, yDataPerPixel } = useDataPerPixel();
+
+  const nLevels = api.levelsSize.length;
+  const levelsSize = useMemo(() => api.levelsSize, [api]);
+
+  const quality = 1 - clamp(qualityFactor, 0, 1);
+  const levelOfDetail = Math.min(
+    Math.floor(Math.log2(Math.max(1, xDataPerPixel, yDataPerPixel)) + quality),
+    Math.max(nLevels - 1, 0)
+  );
+
+  // progressive selects which levels of detail layers are displayed:
+  // true: lower resolution layers displayed behind the current one
+  // false: only current level of detail layer is displayed
+  const lods = progressive
+    ? range(levelOfDetail, nLevels).reverse()
+    : [levelOfDetail];
+
+  const { colorMap, invertColorMap = false } = colorMapAttrs;
+
+  useFrameRendering();
+
+  return (
+    <group>
```

Comment:
You could get away with `<></>` instead of `<group>` here but up to you

Reply:
Good to know!